### PR TITLE
Clients yearly init

### PR DESCRIPTION
**Pull request by fbertsch**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-1261)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/fenix_derived/clients_yearly_v1/init.sql [hunk 1]
```diff
@@ -1,18 +1,50 @@
...
 PARTITION BY
-  submission_date
+  (submission_date)
 CLUSTER BY
-  sample_id,
-  client_id
+  (sample_id)
 OPTIONS
   (require_partition_filter = TRUE)
 AS
```
Comment (ANich) [inline sql/moz-fx-data-shared-prod/fenix_derived/clients_yearly_v1/init.sql:9]: This is all already in `metadata.yaml` no?

File: sql/moz-fx-data-shared-prod/fenix_derived/clients_yearly_v1/init.sql [hunk 2]
```diff
@@ -1,18 +1,50 @@
...
+WITH base AS (
+  -- There are duplicates now in `baseline_clients_daily` because of the join with `clients_first_seen`,
```
Comment (ANich) [inline sql/moz-fx-data-shared-prod/fenix_derived/clients_yearly_v1/init.sql:11]: Link to PR/issue here?

File: sql/moz-fx-data-shared-prod/fenix_derived/clients_yearly_v1/init.sql [hunk 3]
```diff
@@ -1,18 +1,50 @@
...
+        client_id,
+        submission_date
+      ORDER BY
+        first_seen_date ASC,
+        first_run_date ASC
+    ) AS rn,
+  FROM
+    fenix.baseline_clients_daily
+  WHERE
+    submission_date >= "2021-08-01"
```
Comment (ANich) [inline sql/moz-fx-data-shared-prod/fenix_derived/clients_yearly_v1/init.sql:26]: Is it possible to just read from the last 365 `submission_date`'s here? I think in production the query drops rows older than that
```suggestion
    submission_date >= DATE_SUB(CURRENT_DATE(), INTERVAL 365 DAY)
```

**Review by ANich — APPROVED** [review]
The logic LGTM overall left some thoughts. Would be nice to have a test or two

**Comment by ANich** [comment]
Also a `schema.yaml` would be nice to appease the `deploy-changes-to-stage`

**Comment by fbertsch** [comment]
This causes BQ to barf - closing it out in favor of regular backfills for init.

**Pull request closed without merging**